src/fparser/two/utils.py

```diff
@@ -272,23 +272,52 @@ class Base(ComparableMixin):
                     is either str or FortranReaderBase.
       self.item   - Line instance (holds label) or None.
 
+    :param type cls: the class of object to create.
+    :param string: (source of) Fortran string to parse.
+    :type string: `str` or \
+                  :py:class:`fparser.common.readfortran.FortranReaderBase`
+    :param parent_cls: the parent class of this object.
+    :type parent_cls: `type`
+    :param parent: the parent of this node in the parse tree.
+    :type parent: :py:class:`fparser.two.utils.Base` or NoneType
+
     '''
     # This dict of subclasses is populated dynamically by code at the end
     # of this module. That code uses the entries in the
     # 'subclass_names' list belonging to each class defined in this module.
     subclasses = {}
 
+    def __init__(self, string, parent_cls=None, parent=None):
```

Comment:
As discussed, if the only reason for having `parent` being passed by argument is for testing purposes then I don't think it should be here.

Reply:
Removed.